### ISSUE TITLE
ModelAdmin inspect view does not use short_description 

### ISSUE THREAD

**Issue by lb-** [wagtail/wagtail#4320]
### Issue Summary

In ModelAdmin the [short description](http://docs.wagtail.io/en/v1.13.1/reference/contrib/modeladmin/indexview.html#modeladmin-list-display) is not used when a field is referenced in the [inspect_view](http://docs.wagtail.io/en/v1.13.1/reference/contrib/modeladmin/inspectview.html).

Note: The documentation does not say this will work in inspect view, but it is somewhere between an expected behaviour and an enhancement :)

### Steps to Reproduce

1. Start a new project using the Wagtail Bakery Demo
2. Edit bakerydemo/base/models/models.py - add a new method on the `People` model.
```python
@register_snippet
class People(ClusterableModel):
    # ... all other things
    def get_full_name(self):
        return self.first_name + " " + self.last_name
    get_full_name.short_description = "Full Name"
```

3. Edit bakerydemo/base/wagtail_hooks.py - edit the `PeopleModelAdmin` class.

Enable the inspect view + use the `get_full_name` method in place of first_name & last_name (on listing view also).
```python
class PeopleModelAdmin(ModelAdmin):
    # model, menu_label, menu_icon - keep as is
    list_display = ('get_full_name', 'job_title', 'thumb_image') # replace first_name and last_name
    inspect_view_enabled = True # enable the inspect view
    inspect_view_fields = ('get_full_name', 'job_title', 'thumb_image') # add get_full_name and other fields
```

4. Open the admin and navigate to People (via Bakery Misc on the left hand side)
* /admin/base/people/
* Note that the listing view works as expected, our short_description of `Full Name` is used in the header, not `get_full_name`.
![wagtail - people](https://user-images.githubusercontent.com/1396140/36757953-94067d74-1c5e-11e8-968f-710ae4a8259b.png)

5. Open the inspect view for the first person (move mouse over row and click 'inspect')
* /admin/base/people/inspect/4/

**Expected behaviour**
* The inspect view field titles use the same `short_description` as the listing view.
* It would be fair to assume (from a developer perspective) that this setting is used throughout any representation of that field in ModelAdmin.

**Actual behaviour**
* The inspect view field title shows the original method name: eg. `get_full_name`.
![wagtail - inspecting person](https://user-images.githubusercontent.com/1396140/36757987-b4451ad2-1c5e-11e8-84a9-550380e71d8f.png)

### Potential Solution

Not 100% sure but the Listing View (Index) references `header.text`
https://github.com/wagtail/wagtail/blob/master/wagtail/contrib/modeladmin/templates/modeladmin/includes/result_list.html#L9

Where as the inspect view is not passed in any header but simply the field list and uses `field.label`
https://github.com/wagtail/wagtail/blob/master/wagtail/contrib/modeladmin/templates/modeladmin/inspect.html#L32

### Technical details

* Python 3.4.3
* Django version: 2.0.1
* Wagtail version: 2.0b1
* Browser version: Chrome Version 64.0.3282.186 (Official Build) (64-bit) on MacOS

**Comment by lb-** [comment]
I think this may have been fixed by https://github.com/wagtail/wagtail/pull/5147 (that PR calls out this issue specifically).

**Comment by lb-** [comment]
Can confirm - released a little while ago - 2.6

https://github.com/wagtail/wagtail/pull/5147
5c336f223ed2e985e7d6a3a4278c6a3b0b767af5 & b9e6394bd0ba3daea5ce983b12ea812021a6df28

Works as expected.
